src/js/components/DataTable/index.d.ts

```diff
@@ -12,7 +12,7 @@ export interface DataTableProps {
   data?: {}[];
   groupBy?: string | { property: string, expand: Array<string>, onExpand: ((...args: any[]) => any) };
   onMore?: ((...args: any[]) => any);
-  onSearch?: ((...args: any[]) => any);
+  onSearch?: ((search: string) => void);
```

Comment:
I noticed onSearch work properly with DataTable as is, why do we nee this fix? to eliminate the 'any'?

Reply:
The reason to change it is because when the type is `any`, then it is not placing any restriction on what types can be accepted. This won't cause any error, but it also won't help prevent errors for people using Typescript. From the perspective of Typescript, having `any` is like having no type enforcement at because everything will always pass. By changing it, it will make sure that onSearch is only receiving a string.

Reply:
Thank you and of course. I just wanted to make sure the 'any' is the motivation behind the change rather than ts failure that drove this change.